hack/cmd/teststale/teststale.go

```diff
@@ -0,0 +1,203 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+var (
+	binary  = flag.String("binary", "", "absolute filesystem path to the test binary")
+	pkgPath = flag.String("package", "", "test package import path in the format used in the import statements without the $GOPATH prefix")
+)
+
+type pkg struct {
+	dir          string
+	target       string
+	stale        bool
+	testGoFiles  []string
+	testImports  []string
+	xTestGoFiles []string
+	xTestImports []string
+}
+
+func newCmd(format string, pkgPaths []string) *exec.Cmd {
+	args := []string{
+		"list",
+		"-f",
+		format,
+	}
+	args = append(args, pkgPaths...)
+	cmd := exec.Command("go", args...)
+	cmd.Env = os.Environ()
+	return cmd
+}
+
+func newPkg(path string) (*pkg, error) {
+	format := "Dir: {{println .Dir}}Target: {{println .Target}}Stale: {{println .Stale}}TestGoFiles: {{println .TestGoFiles}}TestImports: {{println .TestImports}}XTestGoFiles: {{println .XTestGoFiles}}XTestImports: {{println .XTestImports}}"
```

Comment:
@thockin recently (a couple of days ago or so) I learned that `go list` has a `-json` flag. I couldn't get to this PR after that. I will just use that flag and Unmarshal the JSON.